src/expression/comparison/eq.ts

```diff
@@ -18,18 +19,29 @@ export class Equal extends Comparison {
    * @param {Evaluable} left Left operand.
    * @param {Evaluable} right Right operand.
    */
-  constructor(...args: Evaluable[]);
-  constructor (left: Evaluable, right: Evaluable) {
-    if (arguments.length !== 2) {
-      throw new Error('comparison expression expects left and right operands')
+  private options: ComparisonOptions;
+  constructor(options: ComparisonOptions, ...args: Evaluable[]);
+  constructor (options: ComparisonOptions, left: Evaluable, right: Evaluable) {
```

Comment:
not a big fan of changing the constructor and removing the arguments length check. Rather, I would offer this direction:

Add `strict` prop on the `Comparison` evaluable base, https://github.com/briza-insurance/illogical/blob/master/src/expression/comparison/index.ts ->

```ts
export abstract class Comparison implements Evaluable {
  type: EvaluableType = EvaluableType.Expression

  /**
   * Compared operands must be in the same data type to be successfully
   * evaluated and compared. E.g. [5, 5] vs [5, '5'].
   *
   * When set to false, the operands could some comparison operators will
   * try to cast types to the same type, e.g. [5, '5'] -> [5, 5] before the
   * evaluation. This may lead to unexpected behavior, should be used with caution.
   */
  public strict = true
```

by default set to true.

And change it to false, in the engine when operators are created as needed.

Reply:
9b2e2c0